### ISSUE TITLE
'to satisfy' has surprising results, fails

### ISSUE THREAD

**Issue by gertsonderby** [erikmueller/unexpected-immutable#1]
I ran into the following case:

``` js
 it('tests Immutable to-satisfy', () => {
    let i = Immutable.fromJS({ stuff: 'foo', more: [1,2,3]});
    return expect(i, 'to satisfy', { stuff: 'foo' });
});
```

Output:

```
      UnexpectedError:
expected Map { stuff: 'foo', more: [ 1, 2, 3 ] } to equal { stuff: 'foo' }
```

Seems like it might be oddness regarding `to satisfy`. Adding the following custom assertion fixes it:

``` js
expect.addAssertion('<Immutable> to satisfy <object>', function (expect, subject, pattern) {
    return expect(subject.toJS(), 'to satisfy', pattern);
});
```

**Comment by papandreou** [comment]
That assertion would have to be defined for `to satisfy` to do the right thing here.

PRing that `addAssertion` snippet should pretty much do the trick. Please include support for types that aren't derived from `object` and the `exhaustively` flag, though:

``` js
expect.addAssertion('<Immutable> to [exhaustively] satisfy <any>', function (expect, subject, pattern) {
    return expect(subject.toJS(), 'to [exhaustively] satisfy', pattern);
});
```

**Comment by erikmueller** [comment]
@papandreou thanks for the snippet but could you elaborate on why we need support for `any` here? Because I would also have gone with `<object>` (which might be not optimal as you stated)

**Comment by gertsonderby** [comment]
I've been expanding on the types and assertions a bit since making my first PR. It has a separate `to satisfy` for `<Immutable>`.

**Comment by erikmueller** [comment]
fixed by #8
